docs/Adding-Software-to-Homebrew.md

```diff
@@ -8,46 +8,139 @@ Before you start, please check the open pull requests for [Homebrew/homebrew-cor
 
-1. Make sure you write a good test as part of your formula. Refer to the [Add a test to the formula](Formula-Cookbook.md#add-a-test-to-the-formula) section of the Cookbook for help with this.
+1. You will now have to develop the boilerplate code from `brew create` into a complete formula. Your main references will be the [Formula Cookbook](Formula-Cookbook.md), similar existing formulae, and the official documentation of the software you're packaging. Be sure to also take note of the Homebrew documentation for writing [Python](Python-for-Formula-Authors.md) and [Node](Node-for-Formula-Authors.md) formulae, if applicable.
 
-1. Try installing your formula using `brew install --build-from-source <formula>`, where *\<formula>* is the name of your formula. If any errors occur, correct your formula and attempt to install it again. The formula installation should finish without errors by the end of this step.
+1. Make sure you write a good test as part of your formula. Refer to the [Add a test to the formula](Formula-Cookbook.md#add-a-test-to-the-formula) section of the Formula Cookbook for help with this.
 
 If you're stuck, ask for help on GitHub or the [Homebrew discussion forum](https://github.com/orgs/Homebrew/discussions). The maintainers are very happy to help but we also like to see that you've put effort into trying to find a solution first.
 
 ### Testing and auditing the formula
 
-1. Run `brew audit --strict --new-formula --online <formula>` with your formula. If any errors occur, correct your formula and run the audit again. The audit should finish without any errors by the end of this step.
+1. Test your formula installation using `brew install --formula --build-from-source $(brew --repository homebrew/core)/Formula/<prefix>/<formula>.rb`, where `<prefix>` is the first letter of your formula's name and `<formula>` is the name of your formula. Correct any errors that occur and repeat the installation until it completes without errors.
+
+1. Additionally, verify the proper uninstallation of the formula by using the command `brew uninstall --formula <formula>`, ensuring it removes the software without issues.
+
+1. Run `brew audit --formula --new <formula>` with your formula. If any errors occur, correct your formula and run the audit again. The audit should finish without any errors by the end of this step.
+
+1. Run `brew style --formula --fix <formula>` to automatically check and correct your formula's conformity to Homebrew's style guidelines.
 
 1. Run your formula's test using `brew test <formula>`. The test should finish without any errors.
 
+Keep in mind that these checks will happen automatically when you submit your pull request. Completing them beforehand not only saves time but also makes the whole process smoother for everyone.
+
+If your application and Homebrew do not work well together, feel free to [file an issue](https://github.com/Homebrew/homebrew-core/issues/new/choose) after checking out open issues.
```

Comment:
I'd recommend opening a discussion here rather than issue.

Reply:
I figured that given there is already a 'chooser' screen that tells people where to go for what, it made more sense to point at that than to try and duplicate things that would inherently get out of sync at some stage:

https://github.com/Homebrew/homebrew-core/issues/new/choose

![image](https://github.com/Homebrew/brew/assets/753891/6af21cb6-620e-40a6-8b1a-943672a1deef)

Reply:
Yeh, I'd ideal just put them to discussions here as we 100% will never want an issue filed for this sort of thing, thanks.

Reply:
Is that also true for Casks? As that was where I originally took/modified this example from.

Reply:
Casks links to this section of the README, which seems fine to leave as is:

- https://github.com/Homebrew/homebrew-cask#reporting-bugs

Formula will be updated to the direct discussions link.

Reply:
Actually, for Casks, see this later review comment discussion:

https://github.com/Homebrew/brew/pull/16327#discussion_r1425409450

Reply:
Fixed in 8cbfbeea5f8be0dd0c8fb5aec3a0cd742119313d